src/Uno.UI/UI/Xaml/Controls/PasswordBox/PasswordBox.cs

```diff
@@ -107,7 +108,13 @@ public string Password
 
 		private void OnPasswordChanged(DependencyPropertyChangedEventArgs e)
 		{
-			SetValue(TextProperty, (string)e.NewValue);
+			var newValue = (string)e.NewValue;
+			if (FeatureConfiguration.PasswordBox.PreventPasswordClearOnFocus && newValue.EndsWith(FeatureConfiguration.PasswordBox.PasswordControlChar))
```

Comment:
This is way too hacky IMO.

Instead of using a "marker" character, can we do the change in `SinglelinTextBoxView` to do the following **before** `InsertText`:

```csharp
			this.EditingChanged -= OnEditingChanged;
			this.EditingDidEnd -= OnEditingChanged;
```

Then back:

```csharp
			this.EditingChanged += OnEditingChanged;
			this.EditingDidEnd += OnEditingChanged;

```

after `DeleteBackward`

Also, what happens if the existing Text length is the same as MaxLength?

I'm worried that InsertText will do nothing, then DeleteBackward will remove a user-typed character.

Reply:
@Youssef1313  I was clear that it was very very hacky!! 

The events removal then attachment got me a little scarry as we could break the whole thing if the events, by some mystical reason, do not get attached on time.

Reply:
I think as everything is on the UI thread, it should be fine?

Reply:
The latest published changes seem to work fine. I am still running a couple of manual tests for different scenarios.

InsertText() + DeleteBackward() is "the only" viable solution based on different iOS/Swift threads.

- MaxLength: Does not affect the fix. 
- PasswordChanged Event: removing the listener before InsertText and re-enabling it removes the extra PasswordChanged event we currently have with the previous solution.
- Password is always kept: 

Your thoughts: @Youssef1313 & @jeromelaban

Reply:
Sounds good! Thanks @ajpinedam 

One possible minor issue though is for VoiceOver which will probably read that a character was inserted and deleted. I don't know if it's fixable or not. Probably not that important for now?

Reply:
Good point @Youssef1313 . Let me add VoiceOver to my tests to validate the behavior.